### PR TITLE
editor: experiment suggester

### PR DESCRIPTION
**Pull request by harunurhan**
* Adds a suggester for accelerator_experiments which suggest experiments
and returns results with payload that is populated with $ref of the
accelerator_experiment.

Signed-off-by: harunurhan <harunurhan17@gmail.com>

I am not sure if I need to add tests for this.

### CODE REVIEW

File: inspirehep/config.py [hunk 1]
```diff
@@ -1140,7 +1145,8 @@ def _(x):
 
 # OAuthclient
 # ===========
-orcid.REMOTE_APP['params']['request_token_params'] = {'scope': '/orcid-profile/read-limited /activities/update /orcid-bio/update'}
+orcid.REMOTE_APP['params']['request_token_params'] = {
+    'scope': '/orcid-profile/read-limited /activities/update /orcid-bio/update'}
```
Comment (jacquerie) [inline inspirehep/config.py]: Unrelated, will split.

File: inspirehep/modules/records/receivers.py [hunk 2]
```diff
@@ -371,3 +372,17 @@ def check_if_record_is_going_to_be_deleted(sender, *args, **kwargs):
...
+    accelerator_experiments = json.get("accelerator_experiments")
+    if accelerator_experiments:
+        for accelerator_experiment in accelerator_experiments:
+            experiment = accelerator_experiment.get("experiment")
+            record = accelerator_experiment.get("record")
+            if experiment and record:
+                accelerator_experiment.update({"experiment_suggest": {
+                    "input": experiment,
+                    "payload": {"$ref": record.get("$ref")}
+                }})
```
Comment (jacquerie) [inline inspirehep/modules/records/receivers.py]: Suggestion: every time you write this kind of JSON-driven logic try to see if [`get_value`](https://github.com/inspirehep/inspire-next/blob/2605d9b211d11c58763ba1ce4256f36a7790c01f/inspirehep/utils/record.py#L61-L110) can be used instead.

Comment (jacquerie) [inline inspirehep/modules/records/receivers.py]: BTW this methods looks perfectly testable: you just assert that the right side effects on `json` have happened. You should be able to find examples in `tests/unit/records/test_record_receivers.py` for this.

File: inspirehep/modules/records/mappings/records/hep.json [hunk 3]
```diff
@@ -527,4 +533,4 @@
         "index.percolator.map_unmapped_fields_as_string": true,
         "index.query.default_field": "global_fulltext"
     }
-}
+}
```
Comment (jacquerie) [inline inspirehep/modules/records/mappings/records/hep.json]: Sign that you didn't run `scripts/prettifyjson` : )

**Comment by jacquerie** [comment]
> I am not sure if I need to add tests for this.

Acceptance tests for the entire suggester doing its job will probably have to wait #1730, but unit tests for the receiver can absolutely be written.

**Comment by jmartinm** [comment]
Besides Jacopo's comments, LGTM.

File: inspirehep/config.py (hunk 1 above)
Comment (harunurhan) [inline inspirehep/config.py]: 👍  it was my formatter

File: inspirehep/modules/records/receivers.py [hunk 4]
```diff
@@ -371,3 +372,19 @@ def check_if_record_is_going_to_be_deleted(sender, *args, **kwargs):
...
+    # FIXME: Use a dedicated method when #1355 will be resolved.
+    if "experiments.json" in json.get("$schema"):
+        experiment_names = get_value(json, "experiment_names.title")
+        title_variants = force_force_list(
+            get_value(json, "title_variants.title")
+        )
+        json.update({"experiment_suggest": {
+            "input": experiment_names + title_variants,
+            "output": experiment_names[0],
+            "payload": {"$ref": get_value(json, "self.$ref")},
```
Comment (harunurhan) [inline inspirehep/modules/records/receivers.py]: payload might change later today!

Comment (jacquerie) [inline inspirehep/modules/records/receivers.py]: Did this already happen?

Comment (jmartinm) [inline inspirehep/modules/records/receivers.py]: In the end this is good enough, we don't need extra `payload`.

@harunurhan can you rebase and we can merge? Thanks

**Pull request closed without merging**